conn2res/task.py

```diff
@@ -19,77 +19,120 @@
 from sklearn.metrics import accuracy_score, confusion_matrix, ConfusionMatrixDisplay
 from sklearn.ensemble import RandomForestRegressor
 
-import matplotlib.pyplot as plt
+# import matplotlib.pyplot as plt
 
 
-def check_xy_dims(x,y):
+def check_xy_dims(x, y):
     """
     Check that X,Y have the right dimensions
-    #TODO
+    # TODO
     """
     x_train, x_test = x
     y_train, y_test = y
 
     if ((x_train.ndim == 1) and (x_test.ndim == 1)):
         x_train = x_train[:, np.newaxis]
-        x_test  = x_test[:, np.newaxis]
+        x_test = x_test[:, np.newaxis]
     elif ((x_train.ndim > 2) and (x_test.ndim > 2)):
         x_train = x_train.squeeze()
-        x_test  = x_test.squeeze()
+        x_test = x_test.squeeze()
 
     y_train = y_train.squeeze()
-    y_test  = y_test.squeeze()
+    y_test = y_test.squeeze()
 
     return x_train, x_test, y_train, y_test
 
 
-def regression(x, y, **kwargs):
+def regression(x, y, model=None, **kwargs):
     """
     Regression tasks
-    #TODO
+    # TODO
     """
 
+    # pop variables from kwargs
+    sample_weight_train, sample_weight_test = kwargs.pop(
+        'sample_weight', (None, None))
+
     x_train, x_test = x
     y_train, y_test = y
 
-    model = Ridge(fit_intercept=False, alpha=0.5, **kwargs).fit(x_train, y_train)
-    score = model.score(x_test, y_test)
+    # specify default model
+    if model is None:
+        model = Ridge(fit_intercept=False, alpha=0.5,
+                      **kwargs)
+
+    # fit model on training data
+    model.fit(x_train, y_train, sample_weight_train)
+
+    # calculate model scores on test data
+    score = model.score(x_test, y_test, sample_weight_test)
 
     return score
 
 
-def multiOutputRegression(x, y, **kwargs):
+def multiOutputRegression(x, y, model=None, **kwargs):
     """
     Multiple output regression tasks
-    #TODO
+    # TODO
     """
 
+    # pop variables from kwargs
+    sample_weight_train, sample_weight_test = kwargs.pop(
+        'sample_weight', (None, None))
+
     x_train, x_test = x
     y_train, y_test = y
 
-    model = MultiOutputRegressor(Ridge(fit_intercept=False, alpha=0.5, **kwargs)).fit(x_train, y_train)
+    # specify default model
+    if model is None:
+        model = MultiOutputRegressor(
+            Ridge(fit_intercept=False, alpha=0.5, **kwargs))
+
+    if isinstance(model, MultiOutputRegressor):
+        # fit model on training data
+        model.fit(x_train, y_train)
 
-    y_pred = model.predict(x_test)
-    n_outputs = y_pred.shape[1]
+        # calculate model scores on test data
+        y_pred = model.predict(x_test)
+        n_outputs = y_pred.shape[1]
+        score = []
+        for output in range(n_outputs):
+            score.append(
+                np.abs((np.corrcoef(y_test[:, output], y_pred[:, output])[0][1])))
```

Comment:
I kept this chunk of code as default for backward compatibility. However I would like to point out that the calculation of score in ridge regression is rather different from the implementation here (i.e. calculating the avarege correlation between the predicted and true outputs), so again it is inconsistent with other cases where you use calculate score of the ridge regression/classifier model.